### PR TITLE
add script to create kubernetes_asyncio python client

### PR DESCRIPTION
**Pull request by tomplus**
Script and configuration needed to create the new python asynchronous client. More information here: https://github.com/kubernetes-client/python/issues/323

### CODE REVIEW

File: openapi/python-asyncio.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
```
Comment (brendandburns) [inline openapi/python-asyncio.sh]: nit: 2018

File: openapi/python-asyncio.sh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ARGC=$#
+
+if [ $# -ne 2 ]; then
+    echo "Usage:"
+    echo "    python.sh OUTPUT_DIR SETTING_FILE_PATH"
```
Comment (brendandburns) [inline openapi/python-asyncio.sh]: nit: `python-asyncio.sh`

File: openapi/python-asyncio.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+find "${OUTPUT_DIR}" -path "${OUTPUT_DIR}/base" -prune -o -type f -a -name \*.md -exec sed -i 's/\bclient/kubernetes_asyncio.client/g' {} +
+find "${OUTPUT_DIR}" -path "${OUTPUT_DIR}/base" -prune -o -type f -a -name \*.md -exec sed -i 's/kubernetes_asyncio.client-python/client-python/g' {} +
+
+# workaround https://github.com/swagger-api/swagger-codegen/pull/7905
+find "${OUTPUT_DIR}/client" -type f -name \*.py ! -name '__init__.py' -exec sed -i '/^from .*models.*/d' {} \;
+
+# workaround https://github.com/swagger-api/swagger-codegen/pull/8204
+# + closing session
+# + support application/strategic-merge-patch+json
+echo '21a22,23
```
Comment (brendandburns) [inline openapi/python-asyncio.sh]: Do you really want to inline the patch here? I think it'd be cleaner to check in the patch file.

File: openapi/python-asyncio.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+>             logger.debug("response body: %s", r.data)
+171,172c181,182
+<         if not 200 <= r.status <= 299:
+<             raise ApiException(http_resp=r)
+---
+>             if not 200 <= r.status <= 299:
+>                 raise ApiException(http_resp=r)' | patch "${OUTPUT_DIR}/client/rest.py"
+
+# fix imports
+find "${OUTPUT_DIR}/client/" -type f -name \*.py -exec sed -i 's/import client\./import kubernetes_asyncio.client./g' {} +
```
Comment (brendandburns) [inline openapi/python-asyncio.sh]: There's a certain amount of duplication here, not sure if it's worth extracting into a shared file, but worth considering...

**Comment by brendandburns** [comment]
Sorry for the delay here, some comments, should be straight-forward to get this checked in.

File: openapi/python-asyncio.sh (hunk 4 above)
Comment (tomplus) [reply openapi/python-asyncio.sh]: I can roll these lines to one-liner with gawk probably but for me it's more readable now and I can easily execute it step-by-step. `python.sh` has similar find/sed commands but with different name of package.

File: openapi/python-asyncio.sh (hunk 3 above)
Comment (tomplus) [reply openapi/python-asyncio.sh]: Not really, moved to separated file.

**Comment by brendandburns** [comment]
LGTM.

**Comment by brendandburns** [comment]
Merging.  cc @mbohlool in case he has any concerns, we can always revert.

**Pull request merged**